runtime/autoload/provider/node.vim

```diff
@@ -22,26 +22,27 @@ function! s:is_minimum_version(version, min_major, min_minor) abort
     \         && str2nr(v_list[1]) >= str2nr(a:min_minor)))
 endfunction
 
-function! s:find_node_client(package_manager) abort
-  if !executable(a:package_manager)
-    return ''
-  endif
-  let is_yarn = a:package_manager ==# 'yarn'
-  let cmd = is_yarn ? 'yarn global dir' : 'npm --loglevel silent root -g'
-  let global_modules_dir = get(split(system(cmd), "\n"), 0, '')
-  if v:shell_error || !isdirectory(global_modules_dir)
-    return ''
-  endif
-  " `yarn global dir` returns the parent of '/node_modules'.
-  let global_modules_dir = is_yarn ? global_modules_dir . '/node_modules' : global_modules_dir
-  if !isdirectory(global_modules_dir)
-    return ''
-  endif
-  let entry_point = global_modules_dir . '/neovim/bin/cli.js'
-  if !filereadable(entry_point)
-    return ''
-  endif
-  return entry_point
+let s:NodeHandler = {}
+let s:NodeHandler.result = ''
+function! s:NodeHandler.on_stdout(job_id, data, event)
```

Comment:
You can add `yarn_opts.stdout_buffered = v:true`, then this function won't be needed, and `self.stdout` will become available in `on_exit` (same format as `data`). Analogusly for stderr, if you want to check it.

Reply:
Thanks, @bfredl. I didn't know about that, perhaps the documentation at https://neovim.io/doc/user/job_control.html can be updated to reflect that.

Reply:
It is mentioned at `jobstart()`, which links to `channel.txt`. Perhaps `job_control.txt` should be just merged with `channel.txt`.